modules/tinymce/src/core/main/ts/SelectionOverrides.ts

```diff
@@ -309,13 +309,29 @@ const SelectionOverrides = (editor: Editor): SelectionOverrides => {
         const caretPosition = CaretUtils.getNormalizedRangeEndPoint(dir, rootNode, range);
 
         const beforeNode = caretPosition.getNode(!forward);
-        if (isFakeCaretTarget(beforeNode)) {
-          return showCaret(dir, beforeNode, forward ? !caretPosition.isAtEnd() : false, false);
+        if (Type.isNonNullable(beforeNode)) {
+          if (isFakeCaretTarget(beforeNode)) {
+            return showCaret(dir, beforeNode, forward ? !caretPosition.isAtEnd() : false, false);
+          }
+          if (CaretContainer.isCaretContainer(beforeNode) && NodeType.isContentEditableFalse(beforeNode.nextSibling)) {
```

Comment:
Maybe swap this to `isCaretContainerInline` to be on the safe side so that it only checks for those zwnbsp text nodes not block containers.